src/features/registry.js

```diff
@@ -79,3 +80,11 @@ systemJSPrototype.entries = function () {
     [iterator]: function() { return this }
   };
 };
+
+systemJSPrototype.resolve = function (id, parentUrl) {
```

Comment:
Out of curiosity, why did you move this from import-map.js to registry.js?

Reply:
Good question. It was somewhat arbitrary, I was trying to keep logic related to the registry/import map in their respective files. I wasn't really happy with it but it _felt_ slightly better than doing the check in the import-map file.

I'm happy to change it if the behavior I identified is actually a bug and worth fixing.

Reply:
👍 I don't think there's a bug with it -- my understanding is that system-core.js is a barebones implementation that requires something else to implement a `resolve()` function for it. The risk of moving the resolve function from one file to another would be that not all builds of systemjs include the same files (see [s.js](https://github.com/systemjs/systemjs/blob/master/src/s.js) and [system.js](https://github.com/systemjs/systemjs/blob/master/src/system.js), and not all builds support import-maps.

The `s.js` small build doesn't support import maps, but it also doesn't include `registry.js` as part of its build. So I think all places that use `registry.js` now use import maps and that this won't cause any bugs.

Will defer of course to Guy Bedford for the more definitive answer on this